### PR TITLE
Moved redhat_register snippet above puppet snippet

### PR DESCRIPTION
**Pull request by san7ket**

### CODE REVIEW

File: provisioning_templates/cloud_init/cloud_init_default.erb [hunk 1]
```diff
@@ -24,10 +24,10 @@ hostname: <%= @host.name %>
 fqdn: <%= @host %>
 manage_etc_hosts: true
 users: {}
+<%= snippet 'redhat_register' %>
```
Comment (timogoebel) [inline provisioning_templates/cloud_init/cloud_init_default.erb]: This needs to be inside the
```
runcmd:
- |
```
block, which is specified below.

**Review by timogoebel — CHANGES_REQUESTED** [review]

File: provisioning_templates/cloud_init/cloud_init_default.erb (hunk 1 above)
Comment (san7ket) [reply provisioning_templates/cloud_init/cloud_init_default.erb]: Fixed

**Comment by san7ket** [comment]
Tested the changes on a RHEL box, 
```
2019-08-27 09:15:59,196 - stages.py[DEBUG]: Running module runcmd (<module 'cloudinit.config.cc_runcmd' from '/usr/lib/python2.7/site-packages/cloudinit/config/cc_runcmd.pyc'>) with frequency once-per-instance
2019-08-27 09:15:59,197 - handlers.py[DEBUG]: start: modules-config/config-runcmd: running config-runcmd with frequency once-per-instance
2019-08-27 09:15:59,197 - util.py[DEBUG]: Writing to /var/lib/cloud/instances/i-bd307a3ec329e10a2c/sem/config_runcmd - wb: [644] 19 bytes
2019-08-27 09:15:59,199 - util.py[DEBUG]: Restoring selinux mode for /var/lib/cloud/instances/i-bd307a3ec329e10a2c/sem/config_runcmd (recursive=False)
2019-08-27 09:15:59,200 - util.py[DEBUG]: Restoring selinux mode for /var/lib/cloud/instances/i-bd307a3ec329e10a2c/sem/config_runcmd (recursive=False)
2019-08-27 09:15:59,200 - helpers.py[DEBUG]: Running config-runcmd using lock (<FileLock using file '/var/lib/cloud/instances/i-bd307a3ec329e10a2c/sem/config_runcmd'>)
2019-08-27 09:15:59,200 - schema.py[DEBUG]: Ignoring schema validation. python-jsonschema is not present
2019-08-27 09:15:59,200 - util.py[DEBUG]: Shellified 2 commands.
2019-08-27 09:15:59,201 - util.py[DEBUG]: Writing to /var/lib/cloud/instances/i-bd307a3ec329e10a2c/scripts/runcmd - wb: [700] 2087 bytes
2019-08-27 09:15:59,201 - util.py[DEBUG]: Restoring selinux mode for /var/lib/cloud/instances/i-bd307a3ec329e10a2c/scripts/runcmd (recursive=False)
2019-08-27 09:15:59,202 - util.py[DEBUG]: Restoring selinux mode for /var/lib/cloud/instances/i-bd307a3ec329e10a2c/scripts/runcmd (recursive=False)
2019-08-27 09:15:59,202 - handlers.py[DEBUG]: finish: modules-config/config-runcmd: SUCCESS: config-runcmd ran successfully
```

**Review by timogoebel — APPROVED** [review]
LGTM.

**Comment by timogoebel** [comment]
Thanks, @san7ket.

**Comment by tbrisker** [comment]
pulled this and the original addition into 1.23 as 3157d8b and 39cbd84, would have been good to open a redmine issue for this so we know to pull it into a release (i just happened to notice it while cherry-picking a different issue).

**Pull request merged**